compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -140,8 +140,11 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
 
   protected def transformSpliceType(body: Tree, splice: Select)(using Context): Tree = {
     val body1 = transform(body)(using spliceContext)
-    val tagRef = getQuoteTypeTags.getTagRef(splice.qualifier.tpe.asInstanceOf[TermRef])
-    ref(tagRef).withSpan(splice.span)
+    if ctx.reporter.hasErrors then
+      splice
```

Comment:
Do we need to transform the `body` for this case?

Reply:
We don't need to. That tree will not be used after the error. We could return `body1.select(...).with span(...)` but it might just complicate the code for no practical reason.